docker/solr/entrypoint.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+SOLR_PORT=${SOLR_PORT:-8983}
+SOLR_VERSION=${SOLR_VERSION:-4.9.1}
+DEBUG=${DEBUG:-false}
+SOLR_CORE=${SOLR_CORE:-core0}
+# Since Solr 5.x
+SOLR_COLLECTION=${SOLR_COLLECTION:-gettingstarted}
+
+is_solr_up(){
+    echo "Checking if solr is up on http://localhost:$SOLR_PORT/solr/admin/cores"
+    http_code=`echo $(curl -s -o /dev/null -w "%{http_code}" "http://localhost:$SOLR_PORT/solr/admin/cores")`
+    return `test $http_code = "200"`
+}
+
+wait_for_solr(){
+    while ! is_solr_up; do
+        sleep 3
+    done
+}
+
+keep_solr_up(){
+    trap "exit" INT
+    while is_solr_up; do
+        echo "Still up!" 
+        sleep 2
+    done
+}
+
+run() {
+    dir_name=$1
+    solr_port=$2
+    solr_core=$3
+    # Run solr
+    echo "Running with folder $dir_name"
+    echo "Starting solr on port ${solr_port}..."
+
+    # go to the solr folder
+    cd $1/example
+
+    java -Djetty.port=$solr_port -Dsolr.solr.home=multicore -jar start.jar &
+
+    wait_for_solr
+    cd ../../
+    echo "Started"
+}
+
+initialize_and_run() {
+	version='3.6.0'
+    dir_name="apache-solr-${version}"
+    dir_conf="conf/"
+    add_core $dir_name $dir_conf $SOLR_CORE "$SOLR_CONFS"
```

Comment:
this initialization could also be done at build time, as it only depends on the config files, which are copied in the image at build time.